src/adapter/src/catalog.rs

```diff
@@ -1440,11 +1455,15 @@ impl CatalogState {
         details: EventDetails,
     ) -> Result<(), Error> {
         let user = session.map(|session| session.user().name.to_string());
-        let occurred_at = match (
-            self.unsafe_mode(),
-            self.system_configuration.mock_audit_event_timestamp(),
-        ) {
-            (true, Some(ts)) => ts.into(),
+
+        // Determine if this is a dev build.
+        let is_dev = session
+            .map(|session| session.vars().build_info().is_dev_version())
+            .unwrap_or(false);
+
+        let occurred_at = match self.system_config().mock_audit_event_timestamp() {
+            // Audit event timestamps may only be controlled on dev builds.
+            Some(ts) if is_dev => ts.into(),
```

Comment:
I don't think it's better that we switch from a CLI flag that we can control in tests to a property about the build method, because that means only some binaries are able to pass tests. Can this be a system var also?

Reply:
I believe the idea behind this feature being "double-flagged" (i.e. requiring its own flag in addition to unsafe mode) was to ensure that this flag could not be set in production and impact our billing.

To mitigate the ability for "only some binaries to pass tests", I've added a `skip-if` to the only testdrive test that uses this flag to not be run if we're not on a dev build--this means all binaries can pass all of the tests.

I believe ensuring that we're not running a release build is the best strategy here, but what would you suggest? Adding an additional var for `I_PROMISE_I_AM_NOT_RUNNING_IN_PROD`? My apprehension is that just becomes the opt-in version of unsafe mode.

Reply:
https://github.com/MaterializeInc/materialize/pull/19008 had a similar problem: we needed a super user role to be created for tests. This got done by adding a CLI flag to do a specific thing. For something like this where we are highly concerned about disallowing bad behavior in prod (just like that PR), maybe the same approach is warranted? It would only be set in certain testdrive/slt executions where this specific thing was being tested.

Reply:
Did this